libs/src/services/solana/SolanaUtils.ts

```diff
@@ -119,45 +108,42 @@ class SolanaUtils {
    * Finds a program address, using both seeds, pubkey, and the derived authority.
    * Return [authority, derivedAddress, and bumpSeeds]
    *
-   * @param {PublicKey} programId
-   * @param {PublicKey} address
-   * @param {Uint8Array} seed
-   * @returns {Promise<[PublicKey, PublicKey, number]>}
    */
-  static async findProgramAddressWithAuthority (programId, address, seed) {
+  static async findProgramAddressWithAuthority(
+    programId: PublicKey,
+    address: PublicKey,
+    seed: Uint8Array
+  ) {
     // Finds the authority account by generating a PDA with the address as a seed
     const [authority] = await SolanaUtils.findProgramAddressFromPubkey(
       programId,
       address
     )
 
     const [derivedAddress, bumpSeed] =
-      await SolanaUtils.findProgramAddressFromPubkey(
-        programId,
-        authority,
-        seed
-      )
-    return [authority, derivedAddress, bumpSeed]
+      await SolanaUtils.findProgramAddressFromPubkey(programId, authority, seed)
+    return [authority, derivedAddress, bumpSeed] as const
   }
 
   /**
    * Converts an eth address hex represenatation to an array of Uint8s in big endian notation
-   * @param {string} ethAddress
-   * @returns {Uint8Array}
+   * @param ethAddress
    */
-  static ethAddressToArray (ethAddress) {
+  static ethAddressToArray(ethAddress: string) {
     const strippedEthAddress = ethAddress.replace('0x', '')
     // Need to pad the array to length 20 - otherwise, hex eth keys starting with '0' would
     // result in truncated arrays, while eth spec is always 20 bytes
-    return Uint8Array.of(
-      ...new BN(strippedEthAddress, 'hex').toArray('be', 20)
-    )
+    return Uint8Array.of(...new BN(strippedEthAddress, 'hex').toArray('be', 20))
   }
 
   // Safely create pubkey from nullable val
-  static newPublicKeyNullable (val) {
-    return val ? new PublicKey(val) : null
+  static newPublicKeyNullable<T extends PublicKeyInitData | null>(
+    val: T
+  ): NullablePublicKey<T> {
+    return val
+      ? (new PublicKey(val) as NullablePublicKey<T>)
+      : (null as NullablePublicKey<T>)
```

Comment:
so actually need to do in the first way, since we have to tell each branch which has different conditional type that it actually applies to the return conditional type. down to hop on chat to discuss if that helps